core/storage/user/gae_models.py

```diff
@@ -110,6 +110,22 @@ class UserSettingsModel(base_models.BaseModel):
     preferred_audio_language_code = ndb.StringProperty(
         default=None, choices=[
             language['id'] for language in constants.SUPPORTED_AUDIO_LANGUAGES])
+    # Displayed name for users on android. They are editable unlike usernames.
+    name = ndb.StringProperty(default=None)
+    # A code associated with profiles to provide a PIN based authentication
+    # within the account on android.
+    pin = ndb.StringProperty(default=None)
+    # List of profile user ids associated with a full user. None for profiles.
+    associated_profile_user_ids = ndb.StringProperty(repeated=True)
+    # Exploration language preferences specified by the user for android.
+    preferred_android_language_codes = ndb.StringProperty(
+        repeated=True,
+        indexed=True,
+        choices=[lc['code'] for lc in constants.SUPPORTED_CONTENT_LANGUAGES])
+    # Audio language preference used for audio translations on android.
```

Comment:
Why do we need new Android-specific language codes? Why not use the ones already defined for accounts?

Reply:
Once @vinitamurthi "suggested" me about this while designing the doc that keeping a separate field will allow users to have separate preferences for android and web.
WDYT about this @BenHenning?

Reply:
Hey @sajalasati , just to make it clear: That comment was only to show the trade-off between keeping them separate vs keeping them together. I think its important to understand the trade-off and then make a decision.

Reply:
I added a comment for them saying why I am including it. Trade-offs are just whether we want to provide separate preference choices on web and android or not. 
Although now it does feel more like a prod design concern to me. But apart from that, do you see any other issues that I might be face if I go with the current approach?

Reply:
My main concern is that it introduces complexity since we now have to keep two separate sets of properties in sync. Each time someone adds code to read/update this property, they need to think through why it's potentially different between web & Android (which it may not be). That adds maintenance complexity, and potentially a non-trivial amount.

Reply:
Based on the email discussion, I think we can share the audio language preference with web

Reply:
Done.